tests/fsharpqa/Source/Conformance/InferenceProcedures/NameResolution/RequireQualifiedAccess/OnRecordVsUnion.fs

```diff
@@ -0,0 +1,11 @@
+// #Conformance #TypeInference #Attributes 
+// Verify the RequireQualifiedAccess attribute works on unions
+
+module A =
+    [<RequireQualifiedAccess>]
+    type U = | C
+
+    type C() =
+        static member M() = ()
+
+let x = A.C.M()
```

Comment:
@forki We need to make this change non-breaking. It seems to me you should first search non-RQA unions (emitting no warning), then type names, then RQA unions (emitting the warning)?